tests/preserve_pandas_like_columns_name_attr_test.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pytest
+
+import narwhals.stable.v1 as nw
+
+if TYPE_CHECKING:
+    from tests.utils import Constructor
+
+
+def test_ops_preserve_column_index_name(constructor: Constructor) -> None:
+    if not any(x in str(constructor) for x in ("pandas", "modin", "cudf", "dask")):
+        pytest.skip(
+            reason="Dataframe columns is a list and do not have a `name` like a pandas Index does"
+        )
+
+    data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8.0, 9.0]}
+    df_native = constructor(data)
+    df_native.columns.name = "foo"  # type: ignore[union-attr]
+
+    df = nw.from_native(df_native)
+
+    result = df.with_columns(b=nw.col("a") + 1, c=nw.col("a") * 2).select("c", "b")
```

Comment:
I wanted to concatenate two methods here, mostly for the sake of it.
Might be worth reparametrizing it